File: Sort/Bogo_Sort/bogo_sort.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <random>

bool isSorted(const std::vector<int>& arr) {
    return std::is_sorted(arr.begin(), arr.end());
}

void bogoSort(std::vector<int>& arr) {
    std::random_device rd;
    std::mt19937 g(rd());
    while (!isSorted(arr)) {
        std::shuffle(arr.begin(), arr.end(), g);
    }
}

int main() {
    std::vector<int> data = {3, 1, 4, 2};
    bogoSort(data);
    for (int x : data) std::cout << x << " ";
    return 0;
}

File: Sort/Bogo_Sort/bogo_sort.py
import random

def is_sorted(arr):
    return all(arr[i] <= arr[i+1] for i in range(len(arr)-1))

def bogo_sort(arr):
    while not is_sorted(arr):
        random.shuffle(arr)

if __name__ == "__main__":
    data = [3, 1, 4, 2] # Atenție: n > 8 poate dura "o eternitate"
    bogo_sort(data)
    print(f"Sorted array: {data}")
